--- g10/armor.c
+++ g10/armor.c
@@ -1156,7 +1156,7 @@
 	      }
 
 	    /* write the comment strings */
-	    for(s=comment->d;comment;comment=comment->next,s=comment->d)
+	    for(s=comment?comment->d:NULL;comment;comment=comment->next,s=comment->d)
 	      {
 		iobuf_writestr(a, "Comment: " );
 		for( ; *s; s++ )
--- g10/keylist.c
+++ g10/keylist.c
@@ -1611,8 +1611,8 @@
           char *issuer_fpr = NULL;
           char *reason_text = NULL;
           char *reason_comment = NULL;
-          size_t reason_commentlen;
-          int reason_code;
+          size_t reason_commentlen = 0;
+          int reason_code = 0;
 
 	  if (sig->sig_class == 0x20 || sig->sig_class == 0x28
 	      || sig->sig_class == 0x30)
--- tests/gpgscm/scheme.c
+++ tests/gpgscm/scheme.c
@@ -172,6 +172,7 @@
      case T_FRAME: return "frame";
      }
      assert (! "not reached");
+     return "unknown";
 }
 
 /* ADJ is enough slack to align cells in a TYPE_BITS-bit boundary */
--- sm/certdump.c
+++ sm/certdump.c
@@ -692,9 +692,13 @@
       gpg_err_set_errno (c->error);
       return -1;
     }
-  memcpy (p + c->len, buffer, size);
-  c->len += size;
-  p[c->len] = 0; /* Terminate string. */
+
+  if (p + c->len && buffer)
+    {
+      memcpy (p + c->len, buffer, size);
+      c->len += size;
+      p[c->len] = 0; /* Terminate string. */
+    }
 
   return (gpgrt_ssize_t)size;
 }
--- m4/iconv.m4
+++ m4/iconv.m4
@@ -83,6 +83,7 @@
         size_t res = iconv (cd_utf8_to_88591,
                             (char **) &inptr, &inbytesleft,
                             &outptr, &outbytesleft);
+        iconv_close(cd_utf8_to_88591);
         if (res == 0)
           return 1;
       }
@@ -102,6 +103,7 @@
         size_t res = iconv (cd_88591_to_utf8,
                             (char **) &inptr, &inbytesleft,
                             &outptr, &outbytesleft);
+        iconv_close(cd_88591_to_utf8);
         if ((int)res > 0)
           return 1;
       }
--- configure
+++ configure
@@ -11719,6 +11719,7 @@
         size_t res = iconv (cd_utf8_to_88591,
                             (char **) &inptr, &inbytesleft,
                             &outptr, &outbytesleft);
+        iconv_close(cd_utf8_to_88591);
         if (res == 0)
           return 1;
       }
@@ -11738,6 +11739,7 @@
         size_t res = iconv (cd_88591_to_utf8,
                             (char **) &inptr, &inbytesleft,
                             &outptr, &outbytesleft);
+        iconv_close(cd_88591_to_utf8);
         if ((int)res > 0)
           return 1;
       }
