--- g10/armor.c
+++ g10/armor.c
@@ -1156,7 +1156,7 @@
 	      }
 
 	    /* write the comment strings */
-	    for(s=comment->d;comment;comment=comment->next,s=comment->d)
+	    for(s=comment?comment->d:NULL;comment;comment=comment->next,s=comment->d)
 	      {
 		iobuf_writestr(a, "Comment: " );
 		for( ; *s; s++ )
--- g10/keylist.c
+++ g10/keylist.c
@@ -1611,8 +1611,8 @@
           char *issuer_fpr = NULL;
           char *reason_text = NULL;
           char *reason_comment = NULL;
-          size_t reason_commentlen;
-          int reason_code;
+          size_t reason_commentlen = 0;
+          int reason_code = 0;
 
 	  if (sig->sig_class == 0x20 || sig->sig_class == 0x28
 	      || sig->sig_class == 0x30)
--- tests/gpgscm/scheme.c
+++ tests/gpgscm/scheme.c
@@ -172,6 +172,7 @@
      case T_FRAME: return "frame";
      }
      assert (! "not reached");
+     return "unknown";
 }
 
 /* ADJ is enough slack to align cells in a TYPE_BITS-bit boundary */
