--- src/global.c
+++ src/global.c
@@ -41,8 +41,6 @@
 #include "secmem.h" /* our own secmem allocator */
 
 
-
-
 /****************
  * flag bits: 0 : general cipher debug
  *	      1 : general MPI debug
@@ -79,8 +77,6 @@
 /* Prototypes.  */
 static gpg_err_code_t external_lock_test (int cmd);
 
-
-
 
 /* This is our handmade constructor.  It gets called by any function
    likely to be called at startup.  The suggested way for an
@@ -173,8 +169,6 @@
 }
 
 
-
-
 /* Version number parsing.  */
 
 /* This function parses the first portion of the version number S and
@@ -222,6 +216,7 @@
     return s; /* patchlevel */
 }
 
+
 /* If REQ_VERSION is non-NULL, check that the version of the library
    is at minimum the requested one.  Returns the string representation
    of the library version if the condition is satisfied; return NULL
@@ -297,6 +292,10 @@
                      "clang:" __VERSION__
 #elif __GNUC__
                      "gcc:" __VERSION__
+#elif __SUNPRO_C
+		     "SunC: 0x" #__SUNPRO_C
+#elif __INTEL_COMPILER
+		     "icc: " __INTEL_COMPILER
 #else
                      ":"
 #endif
--- mpi/mpiutil.c
+++ mpi/mpiutil.c
@@ -32,6 +32,21 @@
 static gcry_mpi_t constants[MPI_NUMBER_OF_CONSTANTS];
 
 
+/* Best effort attempt to cleanup constants array */
+#if defined(__GNUC__)
+__attribute__((destructor))
+static void
+constants_cleanup()
+{
+  int idx;
+  for (idx=0; idx < MPI_NUMBER_OF_CONSTANTS; idx++)
+    {
+      constants[idx]->flags &= ~32U;
+      _gcry_mpi_free(constants[idx]);
+    }
+}
+#endif
+
 
 const char *
 _gcry_mpi_get_hw_config (void)
