--- servers/slapd/back-mdb/back-mdb.h
+++ servers/slapd/back-mdb/back-mdb.h
@@ -174,11 +174,7 @@
 #define CACHELINE	64
 #endif
 
-#if defined(__i386) || defined(__x86_64)
-#define MISALIGNED_OK	1
-#else
 #define	ALIGNER	(sizeof(size_t)-1)
-#endif
 
 typedef struct IndexRbody {
 	AttrInfo *ai;
--- libraries/liblmdb/mdb.c
+++ libraries/liblmdb/mdb.c
@@ -184,10 +184,6 @@
 #define BIG_ENDIAN	__BIG_ENDIAN
 #endif
 
-#if defined(__i386) || defined(__x86_64) || defined(_M_IX86)
-#define MISALIGNED_OK	1
-#endif
-
 #include "lmdb.h"
 #include "midl.h"
 
@@ -963,18 +963,14 @@
 	unsigned short *s, *d;	\
 	s = (unsigned short *)&(src);	\
 	d = (unsigned short *)&(dst);	\
-	*d++ = *s++;	\
-	*d++ = *s++;	\
-	*d++ = *s++;	\
-	*d = *s;	\
+	memcpy(d, s, 8);		\
 } while (0)
 #else
 #define COPY_PGNO(dst,src)	do { \
 	unsigned short *s, *d;	\
 	s = (unsigned short *)&(src);	\
 	d = (unsigned short *)&(dst);	\
-	*d++ = *s++;	\
-	*d = *s;	\
+	memcpy(d, s, 4);		\
 } while (0)
 #endif
 #endif
