--- config.h.in
+++ config.h.in
@@ -966,7 +966,9 @@
 
 
 #ifndef UNBOUND_DEBUG
+# ifndef NDEBUG
 #  define NDEBUG
+# endif
 #endif
 
 /** Use small-ldns codebase */
--- configure.ac
+++ configure.ac
@@ -1652,7 +1652,9 @@
 dnl includes
 [
 #ifndef UNBOUND_DEBUG
+# ifndef NDEBUG
 #  define NDEBUG
+# endif
 #endif
 
 /** Use small-ldns codebase */
--- cachedb/cachedb.c
+++ cachedb/cachedb.c
@@ -72,6 +72,17 @@
 #  define be64toh(x) OSSwapBigToHostInt64(x)
 #endif
 
+/* Some compilers do not define __BYTE_ORDER__, like IBM XLC on AIX */
+#if defined(__sun) || defined(_AIX)
+# if __BIG_ENDIAN__
+#  define be64toh(n) (n)
+#  define htobe64(n) (n)
+# else
+#  define be64toh(n) (((uint64_t)htonl((n) & 0xFFFFFFFF) << 32) | htonl((n) >> 32))
+#  define htobe64(n) (((uint64_t)htonl((n) & 0xFFFFFFFF) << 32) | htonl((n) >> 32))
+# endif
+#endif
+
 /** the unit test testframe for cachedb, its module state contains
  * a cache for a couple queries (in memory). */
 struct testframe_moddata {
