--- tests/gpgscm/scheme.c	2018-12-29 16:14:06.000000000 -0500
+++ tests/gpgscm/scheme.c	2018-12-29 16:13:33.000000000 -0500
@@ -872,7 +872,7 @@
 {
 #ifdef NDEBUG
   fprintf(stderr,
-	  "insufficient reservation\n")
+	  "insufficient reservation\n");
 #else
   fprintf(stderr,
 	  "insufficient %s reservation in line %d\n",
@@ -5615,7 +5615,9 @@
   sc->fcells = 0;
   sc->inhibit_gc = GC_ENABLED;
   sc->reserved_cells = 0;
+#ifndef NDEBUG
   sc->reserved_lineno = 0;
+#endif
   sc->no_memory=0;
   sc->inport=sc->NIL;
   sc->outport=sc->NIL;
